homeassistant/components/bluesound/media_player.py

```diff
@@ -634,21 +653,13 @@ def rebuild_bluesound_group(self) -> list[str]:
             if entity.bluesound_device_name in device_group
         ]
 
-    async def async_unjoin(self) -> None:
-        """Unjoin the player from a group."""
-        if self._master is None:
-            return
-
-        _LOGGER.debug("Trying to unjoin player: %s", self.id)
-        await self._master.async_remove_slave(self)
-
-    async def async_add_slave(self, slave_device: BluesoundPlayer) -> None:
+    async def async_add_slave(self, host: str, port: int) -> None:
         """Add slave to master."""
-        await self._player.add_slave(slave_device.host, slave_device.port)
+        await self._player.add_slave(host, port)
 
-    async def async_remove_slave(self, slave_device: BluesoundPlayer) -> None:
+    async def async_remove_slave(self, host: str, port: int) -> None:
         """Remove slave to master."""
-        await self._player.remove_slave(slave_device.host, slave_device.port)
+        await self._player.remove_slave(host, port)
```

Comment:
While we are in the process of touching this code, can we phase out the use of words like slave and master? I know that some are user facing, and maybe we should either deprecate those and make a binary sensor for example

Reply:
Yes I can do that for the HomeAssistant part of the code. 

I'm not entirely sure how to handle it in the library underneath, though. I want to keep it as close to the API as possible, but this might be a worthwhile deviation.

Reply:
What would the naming for that be? **leader/follower**? The media_player documentation uses leader in the context of a group.

Reply:
Sounds awesome